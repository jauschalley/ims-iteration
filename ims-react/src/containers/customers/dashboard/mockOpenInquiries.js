const mockOpenInquiries = {
  "testmode": "0",
  "openInquiry_Count": 7,
  "status_code": "200",
  "inquiries": [
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "March, 05 2020 00:00:00",
          "inquiryDate": "March, 05 2020 00:00:00",
          "assignedByID": 0,
          "state": "",
          "firstName": "",
          "assignedToDate": "March, 05 2020 00:00:00",
          "otherTopic": "",
          "email": "",
          "inquiryID": 56944,
          "otherInformationUse": "",
          "mediumID": 8,
          "orderID": "",
          "customerID": 101656,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "",
          "assignmentNotes": "",
          "addressID": 117658,
          "assignedTo": "Trip Ward",
          "stateID": ""
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "February, 04 2020 00:00:00",
          "inquiryDate": "February, 04 2020 00:00:00",
          "assignedByID": 85,
          "state": "",
          "firstName": "",
          "assignedToDate": "February, 04 2020 00:00:00",
          "otherTopic": "",
          "email": "",
          "inquiryID": 56912,
          "otherInformationUse": "",
          "mediumID": 2,
          "orderID": "",
          "customerID": 101618,
          "assignedBy": "Isabelle Clark",
          "assignedToID": 85,
          "lastName": "",
          "assignmentNotes": "",
          "addressID": 117617,
          "assignedTo": "Isabelle Clark",
          "stateID": ""
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "December, 11 2020 00:00:00",
          "inquiryDate": "December, 11 2019 00:00:00",
          "assignedByID": 50,
          "state": "",
          "firstName": "",
          "assignedToDate": "December, 11 2020 00:00:00",
          "otherTopic": "",
          "email": "",
          "inquiryID": 56334,
          "otherInformationUse": "",
          "mediumID": 1,
          "orderID": 484074,
          "customerID": 101610,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "",
          "assignmentNotes": "",
          "addressID": 117607,
          "assignedTo": "Trip Ward",
          "stateID": ""
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "September, 13 2019 00:00:00",
          "inquiryDate": "September, 13 2019 00:00:00",
          "assignedByID": 50,
          "state": "California",
          "firstName": "Pawfj",
          "assignedToDate": "September, 13 2019 00:00:00",
          "otherTopic": "",
          "email": "hfjfeyatj@dsmmtlsaqnms.cvx",
          "inquiryID": 55305,
          "otherInformationUse": "",
          "mediumID": 4,
          "orderID": 480518,
          "customerID": 100680,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "Gonzales",
          "assignmentNotes": "",
          "addressID": 116441,
          "assignedTo": "Trip Ward",
          "stateID": 6
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "September, 13 2019 00:00:00",
          "inquiryDate": "September, 13 2019 00:00:00",
          "assignedByID": 50,
          "state": "",
          "firstName": "Qkvj",
          "assignedToDate": "September, 13 2019 00:00:00",
          "otherTopic": "",
          "email": "ihkfxuoxto@uamvj.exr",
          "inquiryID": 55304,
          "otherInformationUse": "",
          "mediumID": 1,
          "orderID": 480510,
          "customerID": 41841,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "Ward",
          "assignmentNotes": "",
          "addressID": 50281,
          "assignedTo": "Trip Ward",
          "stateID": ""
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "September, 12 2019 00:00:00",
          "inquiryDate": "September, 12 2019 00:00:00",
          "assignedByID": 50,
          "state": "",
          "firstName": "Tftnvfq",
          "assignedToDate": "September, 12 2019 00:00:00",
          "otherTopic": "",
          "email": "iquhgv.fppbkcco@hiq.sum",
          "inquiryID": 55289,
          "otherInformationUse": "",
          "mediumID": 1,
          "orderID": 480499,
          "customerID": 18666,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "Edwards",
          "assignmentNotes": "",
          "addressID": 22926,
          "assignedTo": "Trip Ward",
          "stateID": ""
      },
      {
          "inquiryComments": "Order to cover 2019 Prev. Res. Gde. Targeted Mailing ",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "February, 19 2019 00:00:00",
          "inquiryDate": "February, 19 2019 00:00:00",
          "assignedByID": 22,
          "state": "Virginia",
          "firstName": "Mltjvadlw",
          "assignedToDate": "February, 19 2019 00:00:00",
          "otherTopic": "",
          "email": "",
          "inquiryID": 52528,
          "otherInformationUse": "",
          "mediumID": 1,
          "orderID": 466884,
          "customerID": 59743,
          "assignedBy": "Stephen Keplinger",
          "assignedToID": 22,
          "lastName": "Targeted-Mailing",
          "assignmentNotes": "",
          "addressID": 69778,
          "assignedTo": "Stephen Keplinger",
          "stateID": 54
      }
  ],
  "inquiry_Count": 7,
  "status_text": "",
  "openInquiries": [
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "March, 05 2020 00:00:00",
          "inquiryDate": "March, 05 2020 00:00:00",
          "assignedByID": 0,
          "state": "",
          "firstName": "",
          "assignedToDate": "March, 05 2020 00:00:00",
          "otherTopic": "",
          "email": "",
          "inquiryID": 56944,
          "otherInformationUse": "",
          "mediumID": 8,
          "orderID": "",
          "customerID": 101656,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "",
          "assignmentNotes": "",
          "addressID": 117658,
          "assignedTo": "Trip Ward",
          "stateID": ""
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "February, 04 2020 00:00:00",
          "inquiryDate": "February, 04 2020 00:00:00",
          "assignedByID": 85,
          "state": "",
          "firstName": "",
          "assignedToDate": "February, 04 2020 00:00:00",
          "otherTopic": "",
          "email": "",
          "inquiryID": 56912,
          "otherInformationUse": "",
          "mediumID": 2,
          "orderID": "",
          "customerID": 101618,
          "assignedBy": "Isabelle Clark",
          "assignedToID": 85,
          "lastName": "",
          "assignmentNotes": "",
          "addressID": 117617,
          "assignedTo": "Isabelle Clark",
          "stateID": ""
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "December, 11 2020 00:00:00",
          "inquiryDate": "December, 11 2019 00:00:00",
          "assignedByID": 50,
          "state": "",
          "firstName": "",
          "assignedToDate": "December, 11 2020 00:00:00",
          "otherTopic": "",
          "email": "",
          "inquiryID": 56334,
          "otherInformationUse": "",
          "mediumID": 1,
          "orderID": 484074,
          "customerID": 101610,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "",
          "assignmentNotes": "",
          "addressID": 117607,
          "assignedTo": "Trip Ward",
          "stateID": ""
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "September, 13 2019 00:00:00",
          "inquiryDate": "September, 13 2019 00:00:00",
          "assignedByID": 50,
          "state": "California",
          "firstName": "Pawfj",
          "assignedToDate": "September, 13 2019 00:00:00",
          "otherTopic": "",
          "email": "hfjfeyatj@dsmmtlsaqnms.cvx",
          "inquiryID": 55305,
          "otherInformationUse": "",
          "mediumID": 4,
          "orderID": 480518,
          "customerID": 100680,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "Gonzales",
          "assignmentNotes": "",
          "addressID": 116441,
          "assignedTo": "Trip Ward",
          "stateID": 6
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "September, 13 2019 00:00:00",
          "inquiryDate": "September, 13 2019 00:00:00",
          "assignedByID": 50,
          "state": "",
          "firstName": "Qkvj",
          "assignedToDate": "September, 13 2019 00:00:00",
          "otherTopic": "",
          "email": "ihkfxuoxto@uamvj.exr",
          "inquiryID": 55304,
          "otherInformationUse": "",
          "mediumID": 1,
          "orderID": 480510,
          "customerID": 41841,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "Ward",
          "assignmentNotes": "",
          "addressID": 50281,
          "assignedTo": "Trip Ward",
          "stateID": ""
      },
      {
          "inquiryComments": "",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "September, 12 2019 00:00:00",
          "inquiryDate": "September, 12 2019 00:00:00",
          "assignedByID": 50,
          "state": "",
          "firstName": "Tftnvfq",
          "assignedToDate": "September, 12 2019 00:00:00",
          "otherTopic": "",
          "email": "iquhgv.fppbkcco@hiq.sum",
          "inquiryID": 55289,
          "otherInformationUse": "",
          "mediumID": 1,
          "orderID": 480499,
          "customerID": 18666,
          "assignedBy": "Trip Ward",
          "assignedToID": 50,
          "lastName": "Edwards",
          "assignmentNotes": "",
          "addressID": 22926,
          "assignedTo": "Trip Ward",
          "stateID": ""
      },
      {
          "inquiryComments": "Order to cover 2019 Prev. Res. Gde. Targeted Mailing ",
          "isStatusClosed": 0,
          "otherInformationType": "",
          "statusChangeDate": "February, 19 2019 00:00:00",
          "inquiryDate": "February, 19 2019 00:00:00",
          "assignedByID": 22,
          "state": "Virginia",
          "firstName": "Mltjvadlw",
          "assignedToDate": "February, 19 2019 00:00:00",
          "otherTopic": "",
          "email": "",
          "inquiryID": 52528,
          "otherInformationUse": "",
          "mediumID": 1,
          "orderID": 466884,
          "customerID": 59743,
          "assignedBy": "Stephen Keplinger",
          "assignedToID": 22,
          "lastName": "Targeted-Mailing",
          "assignmentNotes": "",
          "addressID": 69778,
          "assignedTo": "Stephen Keplinger",
          "stateID": 54
      }
  ]
}

export default mockOpenInquiries;