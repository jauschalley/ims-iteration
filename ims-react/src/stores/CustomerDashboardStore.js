import {observable, decorate} from "mobx";

class CustomerDashboardStore {
  dashboardStatistics = null;
  openInquiries = null;

  constructor(dashboardStatistics = null, openInquiries = null) {
    this.dashboardStatistics = dashboardStatistics || require('../stubs/closed-inquiries.json');
    this.openInquiries = openInquiries || require('../stubs/open-inquiries.json');
  }
}

decorate(CustomerDashboardStore, {
  dashboardStatistics: observable,
  openInquiries: observable
});

export default CustomerDashboardStore;