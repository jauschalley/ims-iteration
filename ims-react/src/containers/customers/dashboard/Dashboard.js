import React from 'react';
import moment from 'moment';
import {observer} from "mobx-react";
import {Row, Col, Container} from 'react-bootstrap';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import CwigCard from '../../../components/CwigCard';
import { tablePaginationStore } from '../../../stores/TablePaginationStore';
import TablePaginationGroup from '../../../components/TablePaginationGroup';

const Dashboard = observer(({customerDashboardStore}) => {
  customerDashboardStore.fetchDashboardStatistics();

  let {week, month, quarter, year} = customerDashboardStore.loading? {} : customerDashboardStore.dashboardStatistics;
  let openInquiries = customerDashboardStore.loading? {} : customerDashboardStore.openInquiries;

  const [page, setPage] = React.useState(0);
  const [rowsPerPage, setRowsPerPage] = React.useState(10);

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = event => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  return(
    customerDashboardStore.loading ? 
    <span>Loading</span> :
    <Container>
      <Row><h1 class="margin-30">Customer Dashboard</h1></Row>
      <CwigCard>
        <Row>
          <h3>Count of Inquiries</h3>
        </Row>
        <Row>
          <Col md="3">
            <p>Week: {week}</p>
          </Col>
          <Col md="3">
            <p>Month: {month}</p>
          </Col>
          <Col md="3">
            <p>Quarter: {quarter}</p>
          </Col>
          <Col md="3">
            <p>Year: {year}</p>
          </Col>
        </Row>
      </CwigCard>
      <CwigCard>
        <Row>
          <h3>Open Inquiries</h3>
        </Row>
        <TableContainer component={Paper}>
          <Table aria-label="open inquiries table">
            <TableHead>
              <TableRow>
                <TableCell>Inquiry Date</TableCell>
                <TableCell>Name</TableCell>
                <TableCell>ID</TableCell>
                <TableCell>Last Action</TableCell>
                <TableCell>Assigned To</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {openInquiries
              .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
              .map(row => (
                <TableRow key={row.name}>
                  <TableCell>{moment(row.inquiryDate).subtract(10, 'days').calendar()}</TableCell>
                  <TableCell>{row.firstName} {row.lastName}</TableCell>
                  <TableCell>{row.inquiryID}</TableCell>
                  <TableCell>{moment(row.statusChangeDate).subtract(10, 'days').calendar()}</TableCell>
                  <TableCell>{row.assignedTo}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
        <TablePaginationGroup tableRef={openInquiries}/>
      </CwigCard>
    </Container>
  );
});

export default Dashboard;