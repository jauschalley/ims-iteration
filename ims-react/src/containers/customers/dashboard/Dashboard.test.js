import React from 'react';
import renderer from 'react-test-renderer';
import Dashboard from './Dashboard';
import CustomerDashboardStore from '../../../stores/CustomerDashboardStore';
import mockClosedInquiries from './mockClosedInquiries';
import mockOpenInquiries from './mockOpenInquiries';

describe('<Dashboard />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Dashboard customerDashboardStore={new CustomerDashboardStore(mockClosedInquiries, mockOpenInquiries)} />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});