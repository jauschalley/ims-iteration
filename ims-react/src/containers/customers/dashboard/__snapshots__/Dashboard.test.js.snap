// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h1>
      Customer Dashboard
    </h1>
  </div>
  <div
    className="cwig-card container"
  >
    <div
      className="row"
    >
      <h3>
        Count of Inquiries
      </h3>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-3"
      >
        <p>
          Week: 
          0
        </p>
      </div>
      <div
        className="col-md-3"
      >
        <p>
          Month: 
          1
        </p>
      </div>
      <div
        className="col-md-3"
      >
        <p>
          Quarter: 
          2
        </p>
      </div>
      <div
        className="col-md-3"
      >
        <p>
          Year: 
          3
        </p>
      </div>
    </div>
  </div>
  <div
    className="cwig-card container"
  >
    <div
      className="row"
    >
      <h3>
        Open Inquiries
      </h3>
    </div>
    <div
      className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="open inquiries table"
        className="MuiTable-root"
      >
        <thead
          className="MuiTableHead-root"
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Inquiry Date
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              ID
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Last Action
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Assigned To
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        >
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              02/24/2020
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              
               
              
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              56944
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              02/24/2020
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Trip Ward
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              01/25/2020
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              
               
              
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              56912
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              01/25/2020
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Isabelle Clark
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              12/01/2019
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              
               
              
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              56334
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              12/01/2020
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Trip Ward
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              09/03/2019
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Pawfj
               
              Gonzales
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              55305
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              09/03/2019
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Trip Ward
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              09/03/2019
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Qkvj
               
              Ward
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              55304
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              09/03/2019
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Trip Ward
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              09/02/2019
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Tftnvfq
               
              Edwards
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              55289
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              09/02/2019
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Trip Ward
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              02/09/2019
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Mltjvadlw
               
              Targeted-Mailing
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              52528
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              02/09/2019
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Stephen Keplinger
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
